lib/puppet/catalog-diff/searchfacts.rb

```diff
@@ -58,20 +59,19 @@ def get_puppetdb_version(connection)
     def find_nodes_puppetdb(env)
       require 'puppet/util/puppetdb'
       server_url = Puppet::Util::Puppetdb.config.server_urls[0]
-      port = server_url.port
-      use_ssl = port != 8080
-      connection = Puppet::Network::HttpPool.http_instance(server_url.host, port, use_ssl)
-      puppetdb_version = get_puppetdb_version(connection)
+      puppetdb_version = get_puppetdb_version(server_url)
       query = build_query(env, puppetdb_version)
       json_query = URI.escape(query.to_json)
+      headers = { 'Accept' => 'application/json'}
+      Puppet.debug("Querying #{server_url} for environment #{env}")
       begin
-        result = connection.request_get("/pdb/query/v4/nodes?query=#{json_query}", 'Accept' => 'application/json')
-        if result.code.to_i >= 400
+        result = Puppet.runtime[:http].get(URI("#{server_url}/pdb/query/v4/nodes?query=#{json_query}"), headers: headers)
+        if result.code >= 400
           puppetdb_version = '2.3'
           Puppet::debug("Query returned HTTP code #{result.code}. Falling back to older version of API used in PuppetDB version #{puppetdb_version}.")
           query = build_query(env, puppetdb_version)
           json_query = URI.escape(query.to_json)
-          result = connection.request_get("/v4/nodes/?query=#{json_query}", 'Accept' => 'application/json')
+          result = Puppet.runtime[:http].get(URI("#{server_url}/pdb/query/v4/nodes?query=#{json_query}"), headers: headers)
```

Comment:
Is there a situation where you'd be using Puppet DB 2.3, and this API would be available??

Reply:
good question. I'm not sure when the API endpoint was introduced. I just copied the old HTTP requests.

Reply:
I think this is fine as it's just the puppet agent running catalog diff that needs to support the new _client_ API.

Reply:
ah you meant the new http client. yes I think that should be fine.